src/drivers/mct/cime_config/buildnml

```diff
@@ -170,21 +169,72 @@ def _create_drv_namelists(case, infile, confdir, nmlgen, files):
     #--------------------------------
     # (1) Write output namelist file drv_in and  input dataset list.
     #--------------------------------
+    write_drv_in_file(case, nmlgen, confdir)
+
+    #--------------------------------
+    # (2) Write out seq_map.rc file
+    #--------------------------------
+    write_seq_maps_file(case, nmlgen, confdir)
+
+    #--------------------------------
+    # (3) Construct and write out drv_flds_in
+    #--------------------------------
+    write_drv_flds_in_file(case, nmlgen, confdir, files)
+
+###############################################################################
+def write_drv_in_file(case, nmlgen, confdir):
+###############################################################################
     data_list_path = os.path.join(case.get_case_root(), "Buildconf", "cpl.input_data_list")
     if os.path.exists(data_list_path):
         os.remove(data_list_path)
     namelist_file = os.path.join(confdir, "drv_in")
     nmlgen.write_output_file(namelist_file, data_list_path )
 
-    #--------------------------------
-    # (2) Write out seq_map.rc file
-    #--------------------------------
+###############################################################################
+def write_seq_maps_file(case, nmlgen, confdir):
+###############################################################################
+    # first determine if there are invalid idmap settings
+    # if source and destination grid are different, mapping file must not be "idmap"
+    gridvalue = {}
+    ignore_component = {}
+    exclude_list = ["CPL","ESP"]
+    for item in case.get_values("COMP_CLASSES"):
+        if item not in exclude_list:
+            gridvalue[item.lower()] = case.get_value(item + "_GRID" )
+            component = case.get_value("COMP_" + item)
+            if case.get_value(item + "_GRID" ) == 'null':
+                ignore_component[item.lower()] = True
+            else:
+                ignore_component[item.lower()] = False
+
+    group_variables = nmlgen.get_group_variables("seq_maps")
+    for name in group_variables:
+        value = group_variables[name]
+        if "type" not in name:
+            value = re.sub('\"', '', value)
+            if 'idmap' == value:
+                component1 = name[0:3]
+                component2 = name[4:7]
+                if not ignore_component[component1]  and not ignore_component[component2]:
+                    if name == "rof2ocn_fmapname" or name == "glc2ice_rmapname":
+                        # TODO: for rof2ocn_fmapname -needs to be resolved since this is
+                        # currently used in prep_ocn_mod.F90 if flood_present is True
+                        logger.warning("   NOTE: ignoring setting of {}=idmap in seq_maps.rc".format(name))
```

Comment:
An issue should be opened for this issue.

Reply:
Ideally, in my mind, we'd have some data-driven list of mapping pairs that are optional (i.e., needed in some configuration but not in others). This can be deferred, though - just mentioning it for the new issue that should be opened.

Reply:
I have opened an issue on this.